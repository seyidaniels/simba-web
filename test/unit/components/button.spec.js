/* eslint-disable no-undef */
// eslint-disable-next-line no-unused-vars
import { mount, shallowMount } from '@vue/test-utils'
import Button from '../../../components/Base/Forms/AppBtn'

describe('Button Component', () => {
  it('shows a button when link prop is absent', () => {
    const wrapper = mount(Button)
    expect(wrapper.contains('button')).toBe(true)
    // expect(wrapper.contains('a')).toBe(false)
  })
  it('shows the appropriate text', () => {
    const wrapper = shallowMount(Button, {
      slots: {
        default: '<div>Login</div>',
      },
    })
    const text = wrapper.find('div').text()
    expect(text).toBe('Login')
  })
  const wrapper = shallowMount(Button, {
    propsData: {
      color: 'primary',
    },
  })
  it('expects color to be primary color', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Expects method to be clicked on button clicked', () => {

  })
  it('Test button  click event', async () => {
    const buttonComponent = {
      components: {
        'app-btn': Button,
      },
      template: `
        <app-btn v-if='!show' @click='mockCallBack'></app-btn>
        <p v-else >button clicked</p>
      `,
      data() {
        return { show: false }
      },
      methods: {
        mockCallBack() {
          this.show = true;
        },
      },
    }
    const wrapper = mount(buttonComponent)
    await wrapper.find('button').trigger('click')
    expect(wrapper.contains('p')).toBe(true)
  })
})
