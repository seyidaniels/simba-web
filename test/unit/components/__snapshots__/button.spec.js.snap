// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component expects color to be the expected theme colors 1`] = `"<v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" depressed=\\"true\\" type=\\"button\\"></v-btn-stub>"`;
